### PR TITLE
Error fix3

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
fixed warnings in our onnx-mlir code; there are still some from deprecated calls in protobuf

also added proper "return false" when any errors are detected in shape inference, which was not systematically done, and made sure also that we emit an error whenever we return false.

### CODE REVIEW

**Comment by tjingrant** [comment]
Hi @AlexandreEichenberger, I recommend the combination of:
- Making failable function return type LogicalResult and,
- Returning emitError upon failure.
To handle helper function failures.

This is the de-facto standard way of handling errors in mlir (e.g., https://github.com/llvm/llvm-project/blob/0752d98ccf8771b41718170d46d11f4020b62818/mlir/lib/Interfaces/ControlFlowInterfaces.cpp#L45). To check if helper function succeeded, we can use `mlir::succeeded/mlir::failed`.

**Comment by AlexandreEichenberger** [comment]
@tjingrant  
I like that; so I will need to change the signature of the shape inference methods, correct?

**Comment by tjingrant** [comment]
I think so.

File: src/Dialect/Krnl/KrnlOps.hpp [hunk 1]
```diff
@@ -32,6 +32,7 @@ class KrnlOpsDialect : public Dialect {
       return LoopType::get(parser.getBuilder().getContext());
 
     parser.emitError(parser.getCurrentLocation(), "Unknown type");
+    return nullptr;
```
Comment (tjingrant) [inline src/Dialect/Krnl/KrnlOps.hpp:35]: i think clang-format bot is complaining about the trailing white space.

Comment (AlexandreEichenberger) [reply src/Dialect/Krnl/KrnlOps.hpp]: tx

**Comment by AlexandreEichenberger** [comment]
did the changes requested with the LogicalResult as requested

**Pull request merged**